### PR TITLE
Fixing execution plan loading

### PR DESCRIPTION
**Pull request by aasimkhan30**
Switching setTimeout loops to message passing to eliminate any race conditions that might be happening. 

Also, added better loading states with support for error state when loading an invalid plan.
Error Page:
image

### CODE REVIEW

File: src/controllers/executionPlanWebviewController.ts [hunk 1]
```diff
@@ -154,4 +165,8 @@ export class ExecutionPlanWebViewController extends ReactWebViewPanelController<
     }
     return sum;
   }
+
+  private updateState() {
```
Comment (laurennat) [inline src/controllers/executionPlanWebviewController.ts:169]: how does this change the state?

File: src/controllers/executionPlanWebviewController.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ import { homedir } from "os";
 import { exists } from "../utils/utils";
 import UntitledSqlDocumentService from '../controllers/untitledSqlDocumentService';
 import * as path from 'path';
+import { ApiStatus } from "../sharedInterfaces/connectionDialog";
```
Comment (kburtram) [inline src/controllers/executionPlanWebviewController.ts:15]: should execution plan take a dependency on connection dialog?  i.e. should we move ApiStatus to a common shared location?

**Review by kburtram — APPROVED** [review]
LGTM, we can move ApiStatus separately if that's easier.

File: src/controllers/executionPlanWebviewController.ts (hunk 2 above)
Comment (aasimkhan30) [reply src/controllers/executionPlanWebviewController.ts:15]: Will move to a new file since it is a common interface

**Review by laurennat — APPROVED** [review]

**Pull request merged**